src/lib/utils.ts

```diff
@@ -0,0 +1,88 @@
+import { createHash } from 'crypto'
+import { resolve } from 'path'
+import PrefixMap from '../routing/prefix-map'
+
+/**
+ * Find the shortest unambiguous prefix of an ILP address in a prefix map.
+ *
+ * This let's us figure out what addresses the selected route applies to. For
+ * example, the most specific route for destination "a.b.c" might be "a", but
+ * that doesn't mean that that route applies to any destination starting with
+ * "a" because there may be a more specific route like "a.c".
+ *
+ * So we would call this utility function to find out that the least specific
+ * prefix for which there are no other more specific routes is "a.b".
+ *
+ * In order to force a minimum prefix, it can be passed as the third parameter.
+ * This function may make it even more specific if necessary to make it
+ * unambiguous, but it will never return a less specific prefix.
+ */
+export const getShortestUnambiguousPrefix = <T> (prefixMap: PrefixMap<T>, address: string, prefix = '') => {
```

Comment:
Shouldn't this method go in the `prefix-map` file, or at least in the routing directory?